tests/ignite/engine/test_engine.py

```diff
@@ -891,3 +891,42 @@ def switch_dataloader():
         trainer.set_data(data2)
 
     trainer.run(data1, max_epochs=10)
+
+
+def test_run_with_max_iters():
+    max_iters = 8
```

Comment:
Could you please add another test with max iters larger than 3 epochs for example, e.g. max_iters=73